src/core/shape_inference/include/nms_shape_inference.hpp

```diff
@@ -284,5 +299,15 @@ std::vector<TRShape> shape_infer(const NonMaxSuppression* op,
     return nms::shape_infer(op, input_shapes, ta, static_output);
 }
 }  // namespace v9
+
+namespace v13 {
+template <class T, class TRShape = result_shape_t<T>>
+std::vector<TRShape> shape_infer(const NMSRotated* op,
+                                 const std::vector<T>& input_shapes,
+                                 const ITensorAccessor& ta = make_tensor_accessor(),
+                                 const bool static_output = !std::is_same<T, PartialShape>::value) {
```

Comment:
To allow GPU implementation reuse the same mechanisms as for NMS, it can be better to keep all of the related `shape_infer` functions in sync. And it can be updated for all nms ops at once when redundant.

Reply:
Yes it should be with sync with common aprocha in shape inference and be compatible with `ShapeInferTA` and `ShapeInferBase`, there are few more types.
The GPU for NMS is like exception and should be kept as exception only. 

Introducing new signature can have consequence as another template has to implemented in shape inference instead using existing one (add new code increase bin size).
This flag is only required in GPU for one version and is was introduced before unification of `shape_infer` functions. The shape inference can force plugin to use its custom implementation on plugin side. The exception should be refactored to be aligned with common approach in shape inference (it was not done because of time). Is better to not introduce new way of shape inference and is better to remove this parameter.

Reply:
Ok, as discussed `static_output` has been removed from the function parameters and resolved by local variable.